assets/frontend/invoices/controllers.js

```diff
@@ -1,16 +1,37 @@
-(function() {
-    'use strict';
+(function () {
+  'use strict';
 
-    angular
-        .module('invoices.portal')
-        .controller('InvoicesController', InvoicesController);
+  angular
+    .module('invoices.portal')
+    .controller('InvoiceController', InvoiceController)
+  ;
 
+   /* INVOICE CONTROLLER: this will handle the data of the invoice lists
+     from the service
+    */
+  function InvoiceController($scope, InvoiceService, $stateParams) {
+    var self = this;
+    self.invoiceService = InvoiceService;
+    var invID = $stateParams.id;
 
-    /* INVOICE CONTROLLER
-     */
-    function InvoicesController($scope) {
-        var self = this;
+    if (typeof invID == 'undefined') {
```

Comment:
this is incorrect. a variable's type becomes undefined if it doesn't exist. You defined the InvID variable by initializing it at line 15.

if you test it, `typeof(invID)` result will be `object`

Also `invId` not `invID`